examples/model_evaluation/plot_feature_importance.py

```diff
@@ -0,0 +1,551 @@
+    engineered_ridge,
+    X_train=X_train,
+    X_test=X_test,
+    y_train=y_train,
+    y_test=y_test,
+)
+comparator = ComparisonReport(reports=[ridge_report, engineered_ridge_report])
+comparator.metrics.report_metrics()
+
+# %%
+# We get a much better score!
+# However, interpreting the features is harder: indeed, our complex feature engineering
+# introduced a lot of features:
+
+# %%
+print("Initial number of features:", X_train.shape[1])
+
+X_train_transformed = engineered_ridge_report.estimator_[:-1].transform(X_train)
+n_features_engineered = X_train_transformed.shape[1]
+print("Number of features after feature engineering:", n_features_engineered)
+
+# %%
+# Let us display the 15 largest absolute coefficients:
+
+# %%
+df_engineered_ridge_coef = engineered_ridge_report.feature_importance.coefficients()
+
+
+def sort_absolute_values(df):
+    df = df.assign(absolute_coefficient=df["Coefficient"].abs()).sort_values(
+        by="absolute_coefficient", ascending=False
+    )
+    return df.head(15)
+
+
+sort_absolute_values(df_engineered_ridge_coef)
+
+# %%
+# We can observe that the most importance features are interactions between several
+# features, that a simple linear model without feature engineering could not have
+# captured.
+
+# %%
+# Compromising on complexity
+# --------------------------
+
+# %%
+# Now, let us build a model with a more interpretable feature engineering, although
+# it might not perform as well.
+# For that, after the complex feature engineering, we perform some feature selection
+# using a :class:`~sklearn.feature_selection.SelectKBest`.
+# To compensate the drop in score, we fine-tune some hyperparameters using
+# a :class:`~sklearn.model_selection.RandomizedSearchCV`.
+
+# %%
+from scipy.stats import randint
+from sklearn.feature_selection import SelectKBest, VarianceThreshold
+from sklearn.model_selection import RandomizedSearchCV
+
+preprocessor = make_column_transformer(
+    (KMeans(n_clusters=20), geo_columns),
+    remainder=SplineTransformer(),
+)
+model = make_pipeline(
+    preprocessor,
+    PolynomialFeatures(degree=1, interaction_only=True, include_bias=False),
+    VarianceThreshold(),
+    SelectKBest(k=50),
+    Ridge(),
+).set_output(transform="pandas")
+
+parameter_grid = {
+    "columntransformer__kmeans__n_clusters": randint(low=10, high=30),
+    "columntransformer__remainder__degree": randint(low=1, high=4),
+    "columntransformer__remainder__n_knots": randint(low=2, high=10),
+    "selectkbest__k": randint(low=5, high=100),
+    "ridge__alpha": np.logspace(-5, 5, num=100),
+}
+random_search = RandomizedSearchCV(
+    model,
+    param_distributions=parameter_grid,
+)
+random_search.fit(X_train, y_train)
+
+selectkbest_ridge = random_search.best_estimator_
+selectkbest_ridge
+
+# %%
+# Let us get the metrics for the best model of our grid search, and compare it with
+# our previous iterations:
+
+# %%
+selectk_ridge_report = EstimatorReport(
+    selectkbest_ridge,
```

Comment:
This should work directly

```suggestion
    random_search,
```